Various improvements + Updated tests

vladrusu opened the pull request:
- Better emoticon handling
- Better clickable & email handling. For simplified logic, replace is done now before parse.
- Ability to support both attributes surrounded by quota or not in non-strict mode for better integration with various BB code editors (like SCEditor)
- Added "decoda-emoticon" class to emoticons in order to be able to style them in code
- Added "decoda-image" class to &lt;img&gt;s in order to be able to style them in code

vladrusu commented:
Hello !
I put all the improvements I made in one big PR.
Hope it suits you! ;-)
Regards,
Vlad

milesj commented on tests/Decoda/Hook/EmoticonHookTest.php:
```diff
@@ -28,13 +28,6 @@ protected function setUp() {
         $this->object->addFilter(new DefaultFilter());
 
         $hook = new EmoticonHook();
-        $hook->addLoader(new DataLoader(array(
```
Reason for this removal?

milesj commented:
Great work so far, thanks for the improvements!

Just one minor question.

vladrusu commented:
Hello !
Thanks!
I added back the custom emoticons tests (sorry, my mistake), with minor adjustments.
Also, I added a test for the unicode emoticon ☺ ... which initially missed from your test.

milesj commented:
Sweet, thanks a lot for the improvements!
